docs/ibis-for-dplyr-users.ipynb

```diff
@@ -0,0 +1,774 @@
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ibis.options.interactive = True"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Loading example data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In R, datasets are typically lazily loaded with packages. For instance, the `starwars` dataset is packaged with dplyr, but is not loaded in memory before you start using it. Ibis provides many datasets in the `examples` module. So to be able to use the `starwars` dataset, you can use:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "starwars = ibis.examples.starwars.fetch()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Similar to dplyr, if we evaluate the name of a table, we get a nicely-formatted table:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "starwars"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Inspecting the dataset with `head()`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Just like in R, you can use `head()` to inspect the beginning of a dataset. You can also specify the number of rows you want to get back by using the parameter `n` (default `n = 5`).\n",
+    "\n",
+    "In R:\n",
+    "\n",
+    "```r\n",
+    "head(starwars) # or starwars |> head()\n",
+    "```\n",
+    "\n",
+    "With Ibis:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "starwars.head(6)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There is no `tail()` in Ibis because most databases do not support this operation.\n",
+    "\n",
+    "Another method you can use to limit the number of rows returned by a query is `limit()` which also takes the `n` parameter."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "starwars.limit(3)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Filtering rows with filter()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Ibis, like dplyr, has the `filter` method to select rows based on conditions.\n",
+    "\n",
+    "With dplyr:\n",
+    "\n",
+    "```r\n",
+    "starwars |>\n",
+    "  filter(skin_color == \"light\")\n",
+    "```\n",
+    "\n",
+    "In Ibis:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "starwars.filter(_.skin_color == \"light\")"
```

Comment:
This is the first place here where Ibis's underscore is used. I think it would be good to briefly explain why it's used. Maybe just something like:
> As mentioned above, Python lacks non-standard evaluation, so we cannot use bare words to refer to column names like we can with dplyr. In Ibis, the columns of a table are represented as attributes of the table object, so you can access them using the syntax `tbl.some_column`. But notice here that we use an underscore (`_`) instead of the table name. The `_` in Ibis represents the table returned by the previous step in the chained sequence of operations.

Reply:
D'oh, I see that this is already explained below :)